src/Npgsql/NpgsqlConnection.cs

```diff
@@ -718,6 +719,13 @@ public async ValueTask DisposeAsync()
         {
             if (_disposed)
                 return;
+
+#if !NET461 && !NETSTANDARD2_0
+            await base.DisposeAsync();
+#else
+            base.Dispose();
+#endif
+
```

Comment:
Let's make it the last operation here since a handler may throw and break the process. The same applies to the sync path as well.